core/src/test/java/com/linecorp/armeria/server/ServerErrorHandlerTest.java

```diff
@@ -18,15 +18,48 @@
 import static org.assertj.core.api.Assertions.assertThat;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
 
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
 import com.linecorp.armeria.common.HttpMethod;
 import com.linecorp.armeria.common.HttpRequest;
 import com.linecorp.armeria.common.HttpResponse;
 import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.SessionProtocol;
 import com.linecorp.armeria.common.annotation.Nullable;
 import com.linecorp.armeria.internal.testing.AnticipatedException;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
 
 class ServerErrorHandlerTest {
+    private static final String TEST_HOST = "foo.com";
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.route()
+              .get("/foo")
+              .errorHandler((ctx, cause) -> null)
+              .build((ctx, req) -> {
+                  throw new RuntimeException();
+              });
+            sb.errorHandler((ctx, cause) -> HttpResponse.of(HttpStatus.BAD_REQUEST));
+        }
+    };
+
+    @ParameterizedTest
+    @CsvSource({ "H1C", "H2C" })
+    void testPassThrough(SessionProtocol protocol) {
```

Comment:
Maybe rename to `shouldFallbackIfServiceErrorHandlerReturnsNull`?
```suggestion
    void shouldFallbackIfServiceErrorHandlerReturnsNull(SessionProtocol protocol) {
```